drivers/dac/ad5770r/ad5770r.c

```diff
@@ -656,6 +684,17 @@ int32_t ad5770r_init(struct ad5770r_dev **device,
 	/* SPI */
 	ret = spi_init(&dev->spi_desc, &init_param->spi_init);
 
+	/* GPIO */
+	ret = gpio_get_optional(&dev->nalarm, init_param->nalarm);
```

Comment:
test for each `ret` and return if non zero

Reply:
this will also fix  the codacy problem

Reply:
Will correct.